doc/guides/index.rst

```diff
@@ -28,3 +28,4 @@ User and Developer Guides
    west/index
    optimizations/index
    zephyr_cmake_package.rst
+   tfm/index
```

Comment:
Looks like these files are mostly in alphabetical order. Why add this at the end?

Reply:
Fixed.